src/data/types.rs

```diff
@@ -281,113 +281,165 @@ pub(super) fn print_type(
     name: Option<InternedStr>,
     f: &mut Formatter,
 ) -> fmt::Result {
-    print_pre(ctype, f)?;
-    print_mid(ctype, name, f)?;
-    print_post(ctype, f)
+    print_declarator(ctype, name, f)?;
+
+    Ok(())
 }
 
-fn print_pre(ctype: &Type, f: &mut Formatter) -> fmt::Result {
-    use Type::*;
-    match ctype {
-        Char(signed) | Short(signed) | Int(signed) | Long(signed) => {
-            let lower = &format!("{:?}", ctype).to_lowercase();
-            let substr = match lower.find('(') {
-                Some(n) => &lower[..n],
-                None => lower.as_str(),
-            };
-            write!(f, "{}{}", if *signed { "" } else { "unsigned " }, substr)
+fn write_struct_type(struct_type: &StructType, f: &mut Formatter) -> fmt::Result {
+    match struct_type {
+        StructType::Named(name, _) => {
+            write!(f, "{}", name)?;
+        }
+        StructType::Anonymous(members) => {
+            write!(f, "{{ ")?;
+            for member in members.iter() {
+                write!(f, "{};", member)?;
+            }
```

Comment:
I know adding proper indentation requires a refactor, but can you add newlines?
```suggestion
            for member in members.iter() {
                writeln!(f, "    {};", member)?;
            }
```

Reply:
Sure.

Reply:
Should I be using four spaces or a tab?

Reply:
Four spaces, same as rust code. Configuring whitespace is a whole different issue haha